packages/interface/src/components/JoinButton.tsx

```diff
@@ -17,12 +17,28 @@ import type { EdDSAPublicKey } from "@pcd/eddsa-pcd";
 import { Button } from "./ui/Button";
 
 export const JoinButton = (): JSX.Element => {
-  const { isLoading, isRegistered, isEligibleToVote, onSignup, gatekeeperTrait, storeZupassProof } = useMaci();
+  const { isLoading, isRegistered, isEligibleToVote, onSignup, gatekeeperTrait, storeZupassProof, zupassProof } =
+    useMaci();
   const signer = useEthersSigner();
   const { address } = useAccount();
 
   const onError = useCallback(() => toast.error("Signup error"), []);
-  const handleSignup = useCallback(() => onSignup(onError), [onSignup, onError]);
+  const handleSignup = useCallback(async () => {
+    // first we call the faucet to get funds
+    const response = await fetch("/api/faucet", {
+      method: "POST",
+      body: JSON.stringify({ address, pcd: zupassProof }),
+      headers: {
+        "Content-Type": "application/json",
+      },
+    });
```

Comment:
I think it should be moved to service layer.
@kittybest moved some logic recently to `server/api` but I don't remember what exactly.

Reply:
Currently, the `fetch` functions are called inside the `utils` folder, and for frontend components to call the api, we put the call inside `server/api` for the components to fetch by calling `api.faucet.getFund` for example.